client/src/components/Header/components/NavLinks.js

```diff
@@ -1,89 +1,166 @@
-import React from 'react';
-import { Link, SkeletonSprite, AvatarRenderer } from '../../helpers';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
-import Login from '../components/Login';
-import { forumLocation } from '../../../../../config/env.json';
-import { useTranslation } from 'react-i18next';
+import { withTranslation } from 'react-i18next';
+import { Link, SkeletonSprite } from '../../helpers';
+import { updateUserFlag } from '../../../redux/settings';
+import {
+  forumLocation,
+  radioLocation,
+  newsLocation,
+  homeLocation,
+  chineseHome
+} from '../../../../../config/env.json';
+
+const {
+  availableLangs,
+  i18nextCodes,
+  langDisplayNames
+} = require('../../../../i18n/allLangs');
+
+const locales = availableLangs.client;
 
 const propTypes = {
   displayMenu: PropTypes.bool,
   fetchState: PropTypes.shape({ pending: PropTypes.bool }),
+  i18n: PropTypes.object,
+  t: PropTypes.func,
+  toggleNightMode: PropTypes.func.isRequired,
   user: PropTypes.object
 };
 
-export function AuthOrProfile({ user, pending }) {
-  const { t } = useTranslation();
-  const isUserDonating = user && user.isDonating;
-  const isUserSignedIn = user && user.username;
-  const isTopContributor =
-    user && user.yearsTopContributor && user.yearsTopContributor.length > 0;
+const mapDispatchToProps = {
+  toggleNightMode: theme => updateUserFlag({ theme })
+};
+
+export class NavLinks extends Component {
+  goToLanguage(lang) {
+    const path = window.location.pathname;
+    switch (lang) {
+      case 'espanol':
+        return `${homeLocation}/espanol${path}`;
+      case 'english':
+        return `${homeLocation}${path}`;
+      case 'chinese':
+        return `${chineseHome}${path}`;
+      default:
+        return `${homeLocation}`;
+    }
+  }
```

Comment:
- [ ] This needs to be created as a utility component (somehere near the createRedirects, etc.)
- [ ] The logic could be made declarative than a ever increasing list of cases.

Reply:
If I'm understanding correctly, I think I've applied this:

```js
const createLanguageRedirect = lang => {
  const path = window.location.pathname;
  if (lang === 'chinese') return `${chineseHome}${path}`;
  if (lang === 'english') return `${homeLocation}${path}`;
  return `${homeLocation}/${lang}${path}`;
};
```

Unless we need to subdomain another language, we shouldn't have to touch this component when we add additional languages to the client.